### PR TITLE
feat(server+client): Update RedesignProject flow 

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Close: #7931

## PR Details

Update ResourceService.redesignProject to use the existing EntityService.createBulkEntitiesAndFields 

There are several use cases that still need to be handled and are not included in this PR:
1. Create new UserAction type for the RedesignProject flow
2. when moving relation fields, the other side is missing some properties like custom attributes, required, serachable, etc
3. When converting a relation field to a scalar field in a target service, set the type of the scalar based on the related ID field.
4. Convert related fields to scalar fields on relation fields of entities that are left in the source service
5. Complete tests for the updated code

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/UserAction/useUserActionWatchStatus.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { useContext, useEffect } from "react";
 import { useQuery } from "@apollo/client";
 
-import * as models from "../../models";
-import { GET_USER_ACTION } from "./queries";
-import { AppContext } from "../../context/appContext";
+import * as models from "../models";
+import { GET_USER_ACTION } from "../Entity/ImportPrismaSchema/queries";
```
Comment (mulygottlieb) [inline packages/amplication-client/src/UserAction/useUserActionWatchStatus.ts:5]: Shouldn't the GET_USER_ACTION query also move out of /Entity/ImportPrismaSchema?

**Review by morhag90 — APPROVED** [review]

File: packages/amplication-server/src/core/resource/resource.service.ts [hunk 2]
```diff
@@ -529,266 +546,282 @@ export class ResourceService {
...
+                await actionContext.onEmitUserActionLog(
+                  `moving relation field ${field.name} to resource ${resourceId} - related entity ${relatedEntityId} is in a different resource`,
+                  EnumActionLogLevel.Info
+                );
+
+                //if the related entity is moved to a different resource - we should remove the relation
+                if (!fieldProperties.allowMultipleSelection) {
+                  createFieldInput.name = `${createFieldInput.name}Id`;
+                  createFieldInput.displayName = `${createFieldInput.displayName} ID`;
+                  createFieldInput.dataType = EnumDataType.SingleLineText; //@todo: set the new type based on the ID type of the related entity (WholeNumber or SingleLineText)
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/resource/resource.service.ts:767]: This @todo (setting the ID type) is potentially a blocker for the beta (@PazYanoverr  @yuval-hazaz )

**Review by mulygottlieb — APPROVED** [review]
Left a few comments worth considering

**Review by mulygottlieb — APPROVED** [review]
Approved additional changes

File: packages/amplication-server/src/core/resource/resource.service.ts (hunk 2 above)
Comment (mulygottlieb) [reply packages/amplication-server/src/core/resource/resource.service.ts:767]: Opened https://github.com/amplication/amplication/issues/7934 to track this. Resolving the comment.

**Pull request merged**